barterdude/mocks.py

```diff
@@ -0,0 +1,88 @@
+import asyncio
+from asynctest.mock import Mock, CoroutineMock
```

Comment:
I'm not confortable of using asynctest as a production dependency... What do you think about this @bluesdog164 ? @leotok can we think about an alternative for this?

Reply:
I had the same feeling about this. I can also change this to use the standard `unittest` lib from python. It doesn't have 
`AsyncMock` in older versions, but it can be implemented. This way we won't have asynctest as a dependecy.
@timotta what do you think?

Reply:
I'm also unconfortable using unittest in production... Maybe we should use this dependency mock as just plain fake objects, what do you think about this?

Reply:
I don't know if @timotta  is not comfortable with using `asynctest` as a dependency in production or mocking production code.

We had this same doubt before. And we could not think in a better solution in short-term. What do you think?

Reply:
We could take advantage of the dynamic nature of python by passing fake objects. So instead of using Mock, MagicMock or CoroutineMock, the developer who use this feature, could create an inline class/object/function and send it to be injected as dependency.

Reply:
You are thinking something in the lines of using ducking type and passing an module that behaves like barterdude. But it would lose the feature of keeping calls registered

Reply:
The problem is exactly what @bluesdog164 mentioned. One of the cool things of this new feature is that is allows developers to see what would happen with a real message. The Mock object is important because it keeps track of what calls and arguments were used during the processing of the message

Example:
```
{
	"message_calls": [
		{
			"method": "accept",
			"args": [],
			"kwargs": {}
		}
	],
	"barterdude_calls": [
		{
			"method": "publish_amqp",
			"args": [],
			"kwargs": {
				"exchange": "NEXT_EXCHANGE_TO_BE_CALLED",
				"data": {
					"list_id": 1055151521,
                    "subject": "vendo samsung galaxy s21",
					"timestamp": 1657231231000
				},
				"properties": {
					"headers": {
						"has_failed": false,
                                                  "trace_id": "random_id"
					}
				}
			}
		}
	]
}
```

Reply:
Yes, in the line of duck typing, but not a module that behaves like barterdude (we are not even mocking barterdude), but modules that behaves like the modules you want to mock. Just like it's implemented, but instead of passing a mock object/function, pass a fake object/function.

Reply:
@leotok publish_amqp is not one mocked by `mock_dependencies` so you have total controll on how to "capture" executions. About the other calls on `mock_dependencies` if we want to track it we can deal with decorators.